tests/simple/simple.go

```diff
@@ -99,9 +99,42 @@ func Match(t *spb.SimpleTest, actual *exprpb.ExprValue) error {
 //	2) Map comparisons ignore order.
 func MatchValue(tag string, expected *exprpb.Value, actual *exprpb.Value) error {
 	// TODO write normalized comparator.
-	// For now, just compare the protos.
-	if !proto.Equal(expected, actual) {
-		return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+	switch expected.GetKind().(type) {
+	case *exprpb.Value_MapValue:
+		// Maps are handled as repeated entries, but the entries need to be
+		// compared using set equality semantics.
+		expectedMap := expected.GetMapValue()
+		actualMap := actual.GetMapValue()
+		if actualMap == nil || expectedMap == nil {
+			return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+		}
+		expectedEntries := expectedMap.GetEntries()
+		actualEntries := actualMap.GetEntries()
+		if len(expectedEntries) != len(actualEntries) {
+			return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+		}
+		for _, expectedElem := range expectedEntries {
+			found := false
+			for _, actualElem := range actualEntries {
+				keyErr := MatchValue(tag, expectedElem.GetKey(), actualElem.GetKey())
+				if keyErr != nil {
+					continue
+				}
+				valErr := MatchValue(tag, expectedElem.GetValue(), actualElem.GetValue())
+				if valErr == nil {
+					found = true
+					break
```

Comment:
This seems like the ideal place for a labeled continue to the outer loop, so you can dispense with the "found" variable.  Also, if the value match is an error, then the entire comparison should be an error (I don't think we support multimaps, right?), so this can be tightened up.